sklearn/utils/estimator_checks.py

```diff
@@ -780,30 +780,16 @@ def check_estimator_sparse_data(name, estimator_orig, strict_mode=True):
             if name in ['Scaler', 'StandardScaler']:
                 estimator.set_params(with_mean=False)
         # fit and predict
-        if "64" in matrix_format:
-            err_msg = (
-                f"Estimator {name} doesn't seem to support {matrix_format} "
-                "matrix, and is not failing gracefully, e.g. by using "
-                "check_array(X, accept_large_sparse=False)"
-            )
-        else:
-            err_msg = (
-                f"Estimator {name} doesn't seem to fail gracefully on sparse "
-                "data: error message should state explicitly that sparse "
-                "input is not supported if this is not the case."
-            )
-        with raises(
-            (TypeError, ValueError),
-            match=["sparse", "Sparse"],
-            may_pass=True,
-            err_msg=err_msg,
-        ):
+        try:
             with ignore_warnings(category=FutureWarning):
                 estimator.fit(X, y)
             if hasattr(estimator, "predict"):
                 pred = estimator.predict(X)
                 if tags['multioutput_only']:
                     assert pred.shape == (X.shape[0], 1)
+                elif tags['multioutput']:
+                    assert (pred.shape == (X.shape[0], 1) or
+                            pred.shape == (X.shape[0],))
```

Comment:
You're right, here I test only with a mono-output (binary) y, so we know the array will be 1D, so I removed these lines. By the way, let me know if you think we should test multioutput estimators with multioutput datasets too. I just devised the test with a binary y for classifiers and a continuous y for regressors, because I just wanted to be able to run the estimator, on sparse and dense inputs and see if they have the same behaviour. But I could do more intensive tests, using several multiclass/multioutput datasets ?